keps/sig-auth/3766-referential-authorization/README.md

```diff
@@ -413,80 +422,115 @@ rejected".
 
 ### API Spec
 
+#### ClusterReferenceConsumer
+
 ```golang
-// ReferenceGrant identifies kinds of resources in other namespaces that are
-// trusted to reference the specified kinds of resources in the same namespace
-// as the policy.
-//
-// Each ReferenceGrant can be used to represent a unique trust relationship.
-// Additional ReferenceGrants can be used to add to the set of trusted
-// sources of inbound references for the namespace they are defined within.
-//
-// ReferenceGrant is a form of runtime verification allowing users to assert
-// which cross-namespace object references are permitted. Implementations that
-// support ReferenceGrant MUST NOT permit cross-namespace references which have
-// no grant, and MUST respond to the removal of a grant by revoking the access
-// that the grant allowed. 
-//
-// Implementation of ReferenceGrant is eventually consistent, dependent on
-// watch events being received from the Kubernetes API. Although some processing
-// delay is inevitable, any updates that could result in revocation of access MUST
-// be considered high priority and handled as quickly as possible.
-//
-// Implementations of ReferenceGrant MUST treat all of the following scenarios
-// as equivalent:
-//
-// * A reference to a Namespace that doesn't exist
-// * A reference to a Namespace that exists and a Resource that doesn't exist
-// * A reference to Namespace and Resource that exists but a ReferenceGrant
-//   allowing the reference does not exist
-//
-// If any of the above occur, a generic error message such as "RefNotPermitted" 
-// should be communicated, likely via status on the referring resource.
-//
-// Support: Core
-type ReferenceGrant struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
+// Subject refers to the subject that is a consumer of the referenced
+// pattern(s).
+type ClusterReferenceConsumer struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Subject refers to the subject that is a consumer of the referenced
+    // pattern(s).
+    Subject Subject       `json:"subject"`
+
+    // From refers to the group and resource that these references originate from.
+    From    GroupResource `json:"from"`
+
+    // To refers to the group and resource that these references target.
+    To      GroupResource `json:"to"`
+
+    // For refers to the purpose of this reference. (Cluster)ReferenceGrants
+    // matching the From, To, and For of this resource will be authorized for
+    // the Subject of this resource.
+    For     string        `json:"for"`
+}
 
-	// Spec defines the desired state of ReferenceGrant.
-	Spec ReferenceGrantSpec `json:"spec,omitempty"`
+// Subject is a copy of RBAC Subject that excludes APIGroup.
+type Subject struct {
+    // Kind of object being referenced. Values defined by this API group are
```

Comment:
Do we leave out `apiGroup` like in RBAC? https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.29/#subject-v1-rbac-authorization-k8s-io

I think we should. But just pointing it out as we already have this kind of struct elsewhere, and there it's relating the "User" and "Group" kinds with the "rbac.authorization.k8s.io" apiGroup (unfortunately, would indeed have been better to deem all those kinds as "core")

Reply:
@liggitt and @deads2k recommended removing apiGroup here based off of learnings from RBAC